lib/literal/types/range_type.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# @api private
+class RangeType < Literal::Type
+	def initialize(type)
+		@type = type
+	end
+
+	def inspect = "_Range(#{@type.inspect})"
+
+	def ===(value)
+		Range === value && (
+			(
+				@type === value.begin && (nil === value.end || @type === value.end)
```

Comment:
Covers a range that begins with the expected type and is either endless or ends with the same type.